app/src/main/java/protect/card_locker/preferences/SettingsActivity.java

```diff
@@ -126,14 +126,24 @@ public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
                 } else {
                     AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM);
                 }
-
+                refreshActivity(true);
```

Comment:
The `refreshActivity` call is very disorienting, as it forces the user back to the top of the screen. So if I scroll down and edit for example "Barcode orientation", the moment I click save I am at the top of settings again instead of staying where I am and seeing the value update.